scenario/lib.py

```diff
@@ -130,10 +132,24 @@ def run(target, test, error_mode, **kwargs):
     print("")
 
   if retcode != 0:
-    scenario.errors = True
-    if error_mode == Err.BREAK:
-      scenario.stop = True
+    scenario.lastRunSucceeded = False
+    if error_mode != Err.IGNORE:
+      scenario.errors = True
+      if error_mode == Err.BREAK:
+        scenario.stop = True
+  else:
+    scenario.lastRunSucceeded = True
 
+def run_and_retry(target, test, retry, **kwargs):
+  """ Run a test and retry it <retry> times if it fails """
+  for iTry in range(1, retry):
+    if iTry == retry:
+      run(target, test, Err.BREAK, **kwargs)
+    else:
+      run(target, test, Err.IGNORE, **kwargs)
```

Comment:
I'd prefer `run` to return a result rathen than the `lastRunSucceeded` mechanism